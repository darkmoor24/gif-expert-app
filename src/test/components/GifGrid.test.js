import React from "react";
// import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Probando componente <GifGrid />', () => {
    const category = "Goku";
    
    test('El componente debe mostrarse correctamente', () => { 
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category }/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargar imagenes useFetchGifs', () => {
        const gifs = [{
            id: "ABC",
            url: "http://example.com/algo.jpg",
            title: "Anything"
        }]; 

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category }/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
    });
});