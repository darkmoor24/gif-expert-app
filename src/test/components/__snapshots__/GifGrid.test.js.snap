// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <GifGrid /> Debe de mostrar items cuando se cargar imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Anything"
      url="http://example.com/algo.jpg"
    />
  </div>
</Fragment>
`;

exports[`Probando componente <GifGrid /> El componente debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
